nilearn/plotting/displays/_slicers.py

```diff
@@ -461,15 +451,36 @@ def _map_show(
         ims = []
         to_iterate_over = zip(self.axes.values(), data_2d_list)
         for display_ax, data_2d in to_iterate_over:
+            # If data_2d is completely masked, then there is nothing to
+            # plot. Hence, no point to do imshow().
             if data_2d is not None:
-                # If data_2d is completely masked, then there is nothing to
-                # plot. Hence, no point to do imshow().
+                data_2d = self._threshold(
+                    data_2d,
+                    threshold,
+                    vmin=kwargs.get("vmin"),
+                    vmax=kwargs.get("vmax"),
+                )
+
                 im = display_ax.draw_2d(
                     data_2d, data_bounds, bounding_box, type=type, **kwargs
                 )
                 ims.append(im)
         return ims
 
+    def _threshold(self, data, threshold=None, vmin=None, vmax=None):
```

Comment:
Yes it applies the same logic (for the same reason as explained [here](https://github.com/nilearn/nilearn/pull/3873#discussion_r1301785684)), though `surf_plotting._threshold` returns the mask and this returns the masked data array. Do we want to refactor them to use the same function? I wasn't sure how best to do that since the surface plotting functions do not seem to use the Nilearn slicers

Reply:
> Do we want to refactor them to use the same function?
At this point I'd say no, your reasoning is good.